### DISCUSSION TITLE
Save HsvColorPicker State?

### DISCUSSION THREAD

**Discussion by j-jmz** [skydoves/colorpicker-compose#49] [Q&A]
Hi Folks, 

I want to use the color picker on a screen to select a color, then leave the screen and when I come back to the color picker screen I want it to have the color i set before. 

Currently I save the picked color in a ViewModel but when I reenter the color picker screen, the color picker reinitializes and overwrites the saved color with a white color.

Is it possible to reenter the color picker screen without reinitializing the HsvColorPicker?

**Comment by faraz152** [comment]
pass your saved color to the `initialColor` parameter when entering the screen:

```kotlin
HsvColorPicker(
    modifier = Modifier.fillMaxWidth().height(300.dp),
    controller = rememberColorPickerController(),
    initialColor = viewModel.savedColor, // restore from viewmodel here
    onColorChanged = { envelope -> viewModel.saveColor(envelope.color) }
)
```

that way the picker starts at the previously selected color instead of defaulting to white.
